tools/waveica/waveica_wrapper.R

```diff
@@ -8,7 +61,14 @@ waveica <- function(data,
                     exclude_blanks) {
 
   # get input from the Galaxy, preprocess data
-  data <- read.csv(data, header = TRUE)
+
+  if (ext == "csv") {
+    data <- read_csv(file, metadata)
+  } else if (ext == "tsv") {
+    data <- read_tsv(file, metadata)
+  } else {
+    data <- read_parquet_file(file, metadata)
+  }
```

Comment:
This constructor can be extracted into a function since it's used in both `waveica` and `waveica_singlebatch`

Reply:
implementation change in [9bee08a](https://github.com/RECETOX/galaxytools/pull/304/commits/9bee08ace9062a191a0b6f43344ff4f078517345)